pkg/openstack/instances.go

```diff
@@ -625,5 +664,64 @@ func getAttachedInterfacesByID(client *gophercloud.ServiceClient, serviceID stri
 		return interfaces, err
 	}
 
+	// Check if trunk ports are attached
+	var subports []trunks.Subport
+
+	listOpts := trunks.ListOpts{}
+	allPages, err := trunks.List(network, listOpts).AllPages()
```

Comment:
will this impact env does not have Trunk support?

Reply:
Same question for me.

Reply:
Hi, 

There is no impact on env that does not have trunk support, ourselves do deployment with two flavors using trunks (SRIOV) and not using trunks (legacy VLAN/VxLAN), that is also the reason why those "acceptance" tests passed in CI (they all don't support trunk?).

This pull request was created before the nodeAddresses() function is moved from openstack.go to instances.go, at that time all acceptance tests are passed. I rebased the pull request to the master, after that, openlab/cloud-provider-openstack-acceptance-test-e2e-conformance is able to pass, but those 3 openlab/cloud-provider-openstack-acceptance-test-e2e-conformance-stable-branch-vXX failed, not sure why.

Please help scrutinize the pull request, comment and approve.

Thanks

Jing